integration-tests/deployment/ccip/add_lane_test.go

```diff
@@ -35,32 +66,90 @@ func TestAddLane(t *testing.T) {
 	// We expect no lanes available on any chain.
 	state, err = LoadOnchainState(e.Env, e.Ab)
 	require.NoError(t, err)
-	for _, chain := range state.Chains {
+	for _, sel := range []uint64{chain1, chain2} {
+		chain := state.Chains[sel]
 		offRamps, err := chain.Router.GetOffRamps(nil)
 		require.NoError(t, err)
 		require.Len(t, offRamps, 0)
 	}
 
-	// Add one lane and send traffic.
-	from, to := e.Env.AllChainSelectors()[0], e.Env.AllChainSelectors()[1]
-	require.NoError(t, AddLane(e.Env, state, from, to))
+	replayBlocks, err := LatestBlocksByChain(testcontext.Get(t), e.Env.Chains)
+	require.NoError(t, err)
+
+	// Add one lane from chain1 to chain 2 and send traffic.
+	require.NoError(t, AddLane(e.Env, state, chain1, chain2))
 
-	for sel, chain := range state.Chains {
+	for _, sel := range []uint64{chain1, chain2} {
+		chain := state.Chains[sel]
 		offRamps, err := chain.Router.GetOffRamps(nil)
 		require.NoError(t, err)
-		if sel == to {
+		if sel == chain2 {
 			require.Len(t, offRamps, 1)
+			srcCfg, err := chain.OffRamp.GetSourceChainConfig(nil, chain1)
+			require.NoError(t, err)
+			require.Equal(t, common.LeftPadBytes(state.Chains[chain1].OnRamp.Address().Bytes(), 32), srcCfg.OnRamp)
 		} else {
 			require.Len(t, offRamps, 0)
 		}
 	}
-	latesthdr, err := e.Env.Chains[to].Client.HeaderByNumber(testcontext.Get(t), nil)
+
+	time.Sleep(30 * time.Second)
+	ReplayLogs(t, e.Env.Offchain, replayBlocks)
+
+	latesthdr, err := e.Env.Chains[chain2].Client.HeaderByNumber(testcontext.Get(t), nil)
 	require.NoError(t, err)
 	startBlock := latesthdr.Number.Uint64()
-	seqNum := SendRequest(t, e.Env, state, from, to, false)
+
+	seqNum := TestSendRequest(t, e.Env, state, chain1, chain2, false)
 	require.Equal(t, uint64(1), seqNum)
-	require.NoError(t, ConfirmExecWithSeqNr(t, e.Env.Chains[from], e.Env.Chains[to], state.Chains[to].OffRamp, &startBlock, seqNum))
+	require.NoError(t,
+		ConfirmCommitWithExpectedSeqNumRange(
+			t, e.Env.Chains[chain1], e.Env.Chains[chain2],
+			state.Chains[chain2].OffRamp, &startBlock,
+			cciptypes.SeqNumRange{
+				cciptypes.SeqNum(seqNum),
+				cciptypes.SeqNum(seqNum),
+			}))
+	require.NoError(t, ConfirmExecWithSeqNr(t, e.Env.Chains[chain1], e.Env.Chains[chain2], state.Chains[chain2].OffRamp, &startBlock, seqNum))
 
-	// TODO: Add a second lane, then disable the first and
-	// ensure we can send on the second but not the first.
+	// Add another lane
+	replayBlocks, err = LatestBlocksByChain(testcontext.Get(t), e.Env.Chains)
+	require.NoError(t, err)
+	require.NoError(t, AddLane(e.Env, state, chain2, chain1))
+
+	// disable onRamp for previous lane chain1 -> chain2
+	updates := []router.RouterOnRamp{
+		{
+			DestChainSelector: chain2,
+			OnRamp:            common.HexToAddress("0x"),
+		},
+	}
+
+	tx, err := state.Chains[chain1].Router.ApplyRampUpdates(e.Env.Chains[chain1].DeployerKey,
```

Comment:
ah this will just cover the router logic which I don't think needs further testing. What I wanted to test here was the offchains code support for ignoring a lane, as in just removing a source from the offramp, sending a message from that source and seeing that it never gets processed but other messages would still be processed for the enabled source.

Reply:
removing a source from offRamp is not possible currently because of this https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.8/ccip/offRamp/OffRamp.sol#L974-L978 -is there any other way to achieve this?

Reply:
Updated based on second point